plutus-scb/src/Control/Monad/Freer/Extra/Log.hs

```diff
@@ -29,8 +36,22 @@ logInfo = Eff.send . Log LevelInfo
 logDebug :: Member Log effs => Text -> Eff effs ()
 logDebug = Eff.send . Log LevelDebug
 
-runStderrLog :: (LastMember IO effs) => Eff (Log ': effs) ~> Eff effs
+logWarn :: Member Log effs => Text -> Eff effs ()
+logWarn = Eff.send . Log LevelWarn
+
+runStderrLog :: (LastMember m effs, MonadIO m) => Eff (Log ': effs) ~> Eff effs
 runStderrLog =
     Eff.interpretM
         (\case
-             Log level txt -> runStderrLoggingT $ logWithoutLoc "" level txt)
+             Log level txt -> liftIO $ runStderrLoggingT $ logWithoutLoc "" level txt)
+
+-- | Re-interpret a 'Writer' effect by writing the events to the log
+writeToLog ::
```

Comment:
Cute. Although I guess a bunch of those could be using `Log` directly? Perhaps we should move `Log` to `plutus-emulator`?

Reply:
Done. Yeah a lot of the `Event` types in `plutus-emulator` are just log messages.